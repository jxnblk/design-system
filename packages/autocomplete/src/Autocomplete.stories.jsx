/* eslint-disable react/no-children-prop */
import React from 'react'
import Component from '@reach/component-component'
import { within, userEvent } from '@storybook/testing-library'
import { expect } from '@storybook/jest'
import us from 'us'

import { Box, Text, ThemeProvider } from 'pcln-design-system'
import { Pin as PinIcon } from 'pcln-icons'

import catNames from 'cat-names'
import { Autocomplete, Label, Input, Menu, Item } from '.'

const usStates = us.STATES.map((state) => (
  <Autocomplete.Item key={state.name} item={state}>
    {state.name}
  </Autocomplete.Item>
))

const cats = catNames.all
const match = (item, value) => item.includes(value)

const kayakTheme = {
  font: "'HelveticaNeue-Bold', 'Helvetica Neue', Helvetica, Arial, sans-serif",
  palette: {
    primary: {
      light: '#ff873f',
      base: '#ff690f',
      dark: '#b64907',
    },
    secondary: {
      light: '#ff873f',
      base: '#ff690f',
      dark: '#b64907',
    },
  },
}

export default {
  title: 'pcln-autocomplete/Autocomplete',
  component: Autocomplete,
}

export const Default = () => (
  <Component
    initialState={{ value: '' }}
    children={({ state, setState }) => (
      <Box>
        <Autocomplete
          onChange={(item) => {
            setState({ value: item })
          }}
          match={match}
        >
          <Label mb={1}>Cat</Label>
          <Input />
          <Menu>
            {cats.map((cat) => (
              <Item key={cat} item={cat}>
                <PinIcon color='primary' mr={2} />
                <Text fontSize={0}>{cat}</Text>
              </Item>
            ))}
          </Menu>
        </Autocomplete>
        <Text my={2}>This text should be covered up by the Autocomplete.Menu when open.</Text>
        <Text my={2}>
          The current value is <code>{state.value}</code>
        </Text>
      </Box>
    )}
  />
)

Default.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)

  expect(canvas.queryByText(/Boots/)).not.toBeInTheDocument()

  const autocompleteInput = canvas.getByLabelText('Cat', {
    selector: 'input',
  })

  await userEvent.type(autocompleteInput, 'Bo', {
    delay: 100,
  })

  expect(canvas.getByText(/Boots/)).toBeInTheDocument()

  await userEvent.hover(canvas.getByText(/Boots/))
}

export const Themed = () => (
  <ThemeProvider theme={kayakTheme}>
    <Component
      initialState={{ value: '' }}
      children={({ state, setState }) => (
        <Box>
          <Autocomplete
            onChange={(item) => {
              setState({ value: item })
            }}
            match={match}
          >
            <Label mb={1}>Cat</Label>
            <Input />
            <Menu>
              {cats.map((cat) => (
                <Item key={cat} item={cat}>
                  <PinIcon color='primary' mr={2} />
                  <Text fontSize={0}>{cat}</Text>
                </Item>
              ))}
            </Menu>
          </Autocomplete>
          <Text my={2}>This text should be covered up by the Autocomplete.Menu when open.</Text>
          <Text my={2}>
            The current value is <code>{state.value}</code>
          </Text>
        </Box>
      )}
    />
  </ThemeProvider>
)

export const DefaultHighlighted = () => (
  <Autocomplete defaultIsOpen={true} defaultHighlightedIndex={3}>
    <Autocomplete.Menu>{usStates}</Autocomplete.Menu>
  </Autocomplete>
)
