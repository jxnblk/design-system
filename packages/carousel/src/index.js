export * from './Carousel'
export * from './ArrowButton'
