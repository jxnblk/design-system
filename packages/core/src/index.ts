import * as storybookArgs from './storybook/args'

export type { IButtonProps } from './Button'
export type { IDialogProps } from './Dialog'
export type { IGridProps } from './Grid'
export type { ICloseButtonProps } from './CloseButton'

export { Absolute } from './Absolute'
export { Accordion } from './Accordion'
export * from './Animate'
export { Avatar } from './Avatar'
export { BackgroundImage } from './BackgroundImage'
export { Badge } from './Badge'
export { Banner } from './Banner'
export { BlockLink } from './BlockLink'
export { Box } from './Box'
export { Breadcrumbs } from './Breadcrumbs'
export { Button } from './Button'
export { Card } from './Card'
export { Checkbox } from './Checkbox'
export { ButtonChip, FilterChip as Chip, ChoiceChip, FilterChip } from './Chip'
export { CloseButton } from './CloseButton'
export * from './Container'
export { Dialog } from './Dialog'
export { Divider } from './Divider'
export { DotLoader } from './DotLoader'
export { Flag } from './Flag'
export { Flex } from './Flex'
export { FormField, InputField } from './FormField'
export { GenericBanner } from './GenericBanner'
export * from './Grid'
export { Heading } from './Heading'
export { Hide } from './Hide'
export { Hug } from './Hug'
export { Icon } from './Icon'
export { IconButton } from './IconButton'
export { IconField } from './IconField'
export { Image } from './Image'
export { Input } from './Input'
export { InputGroup } from './InputGroup'
export { Label } from './Label'
export * from './Layout'
export { Link } from './Link'
export { List } from './List'
export { PasswordInput } from './PasswordInput'
export { PlaceholderImage } from './PlaceholderImage'
export * from './Popout'
export * from './ProgressBar'
export { Radio } from './Radio'
export { RadioCheckToggleCard } from './RadioCheckToggleCard'
export { RatingBadge } from './RatingBadge'
export { Relative } from './Relative'
export { Select } from './Select'
export { ShadowEffect, ShadowOverlay } from './ShadowEffect'
export { Shimmer } from './Shimmer'
export { SkipMenu } from './SkipMenu'
export { SlideBox } from './SlideBox'
export { Spinner } from './Spinner'
export { SrOnly } from './SrOnly'
export { Stamp } from './Stamp'
export { Stepper } from './Stepper'
export { Swatch } from './Swatch'
export { Text } from './Text'
export { TextArea } from './TextArea'
export { Toast } from './Toast'
export { ToastProvider } from './ToastProvider'
export { Toggle } from './Toggle'
export { ToggleBadge } from './ToggleBadge'
export { Tooltip } from './Tooltip'
export { Truncate } from './Truncate'

export { ThemeProvider } from './ThemeProvider'

export { useToast } from './ToastProvider'

export * from './utils'

// DocsUtils
export * from './DocsUtils/DoDont'
export * from './DocsUtils/DocTable'
export * from './DocsUtils/Hero'
// export * from './DocsUtils/LiveDemo'
export * from './DocsUtils/Note'
export * from './DocsUtils/RelatedComponent'
export * from './DocsUtils/Section'
export * from './DocsUtils/StoryHeading'
export * from './DocsUtils/StoryStage'
export * from './DocsUtils/TableOfContents'
export { storybookArgs }

export * from './theme'
