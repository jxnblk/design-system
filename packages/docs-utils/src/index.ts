/* istanbul ignore file */
export * from './components'
export * as utils from './utils'
