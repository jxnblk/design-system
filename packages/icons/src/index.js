export { default as Ac } from './Ac.jsx'
export { default as Accessible } from './Accessible.jsx'
export { default as AddAlarm } from './AddAlarm.jsx'
export { default as Airplane } from './Airplane.jsx'
export { default as Arrival } from './Arrival.jsx'
export { default as ArrowDown } from './ArrowDown.jsx'
export { default as ArrowLeft } from './ArrowLeft.jsx'
export { default as ArrowRight } from './ArrowRight.jsx'
export { default as ArrowUp } from './ArrowUp.jsx'
export { default as Attention } from './Attention.jsx'
export { default as Automatic } from './Automatic.jsx'
export { default as Backpack } from './Backpack.jsx'
export { default as Bag } from './Bag.jsx'
export { default as Beach } from './Beach.jsx'
export { default as Bed } from './Bed.jsx'
export { default as BedKing } from './BedKing.jsx'
export { default as BedSingle } from './BedSingle.jsx'
export { default as BeenHere } from './BeenHere.jsx'
export { default as Bike } from './Bike.jsx'
export { default as Bookmark } from './Bookmark.jsx'
export { default as BookmarkActivated } from './BookmarkActivated.jsx'
export { default as BoxChecked } from './BoxChecked.jsx'
export { default as BoxEmpty } from './BoxEmpty.jsx'
export { default as BoxMinus } from './BoxMinus.jsx'
export { default as BoxPlus } from './BoxPlus.jsx'
export { default as Breakfast } from './Breakfast.jsx'
export { default as Broom } from './Broom.jsx'
export { default as Build } from './Build.jsx'
export { default as Bus } from './Bus.jsx'
export { default as Business } from './Business.jsx'
export { default as Cake } from './Cake.jsx'
export { default as Calendar } from './Calendar.jsx'
export { default as CalendarCheck } from './CalendarCheck.jsx'
export { default as CalendarClock } from './CalendarClock.jsx'
export { default as CarAutomatic } from './CarAutomatic.jsx'
export { default as CarCircle } from './CarCircle.jsx'
export { default as CarDoor } from './CarDoor.jsx'
export { default as CarManual } from './CarManual.jsx'
export { default as CarWash } from './CarWash.jsx'
export { default as CardTravel } from './CardTravel.jsx'
export { default as Carriage } from './Carriage.jsx'
export { default as CarryOnBag } from './CarryOnBag.jsx'
export { default as Cars } from './Cars.jsx'
export { default as Casino } from './Casino.jsx'
export { default as ChargeStation } from './ChargeStation.jsx'
export { default as Chart } from './Chart.jsx'
export { default as ChartAlt } from './ChartAlt.jsx'
export { default as Chat } from './Chat.jsx'
export { default as ChatBubble } from './ChatBubble.jsx'
export { default as Check } from './Check.jsx'
export { default as CheckedBag } from './CheckedBag.jsx'
export { default as Checklist } from './Checklist.jsx'
export { default as ChevronDown } from './ChevronDown.jsx'
export { default as ChevronLeft } from './ChevronLeft.jsx'
export { default as ChevronRight } from './ChevronRight.jsx'
export { default as ChevronUp } from './ChevronUp.jsx'
export { default as CityView } from './CityView.jsx'
export { default as Clock } from './Clock.jsx'
export { default as Close } from './Close.jsx'
export { default as Cloud } from './Cloud.jsx'
export { default as CollisionCoverage } from './CollisionCoverage.jsx'
export { default as ContactMail } from './ContactMail.jsx'
export { default as ContactPhone } from './ContactPhone.jsx'
export { default as ConvenienceStore } from './ConvenienceStore.jsx'
export { default as Copy } from './Copy.jsx'
export { default as Couch } from './Couch.jsx'
export { default as Coupon } from './Coupon.jsx'
export { default as CreditCard } from './CreditCard.jsx'
export { default as Cruises } from './Cruises.jsx'
export { default as Departure } from './Departure.jsx'
export { default as Devices } from './Devices.jsx'
export { default as DevicesImportant } from './DevicesImportant.jsx'
export { default as Directions } from './Directions.jsx'
export { default as Discount } from './Discount.jsx'
export { default as Document } from './Document.jsx'
export { default as Dollar } from './Dollar.jsx'
export { default as DollarCircle } from './DollarCircle.jsx'
export { default as Dot } from './Dot.jsx'
export { default as DoubleBed } from './DoubleBed.jsx'
export { default as DoubleOccupancy } from './DoubleOccupancy.jsx'
export { default as Download } from './Download.jsx'
export { default as DownloadAlt } from './DownloadAlt.jsx'
export { default as EarlyBird } from './EarlyBird.jsx'
export { default as Edit } from './Edit.jsx'
export { default as Electric } from './Electric.jsx'
export { default as Email } from './Email.jsx'
export { default as Emoticon } from './Emoticon.jsx'
export { default as Event } from './Event.jsx'
export { default as EventAvailable } from './EventAvailable.jsx'
export { default as EventBusy } from './EventBusy.jsx'
export { default as ExitToApp } from './ExitToApp.jsx'
export { default as ExperienceTicket } from './ExperienceTicket.jsx'
export { default as ExternalLink } from './ExternalLink.jsx'
export { default as Facebook } from './Facebook.jsx'
export { default as Family } from './Family.jsx'
export { default as FavoriteHotel } from './FavoriteHotel.jsx'
export { default as Filter } from './Filter.jsx'
export { default as Fitness } from './Fitness.jsx'
export { default as Flame } from './Flame.jsx'
export { default as FlightCircle } from './FlightCircle.jsx'
export { default as FlightCoverage } from './FlightCoverage.jsx'
export { default as Flights } from './Flights.jsx'
export { default as FreeCancellation } from './FreeCancellation.jsx'
export { default as Fridge } from './Fridge.jsx'
export { default as Gallery } from './Gallery.jsx'
export { default as Gas } from './Gas.jsx'
export { default as Globe } from './Globe.jsx'
export { default as Golf } from './Golf.jsx'
export { default as GolfTee } from './GolfTee.jsx'
export { default as Gps } from './Gps.jsx'
export { default as Graph } from './Graph.jsx'
export { default as Grid } from './Grid.jsx'
export { default as Group } from './Group.jsx'
export { default as GuestScore } from './GuestScore.jsx'
export { default as Guests } from './Guests.jsx'
export { default as Help } from './Help.jsx'
export { default as History } from './History.jsx'
export { default as Home } from './Home.jsx'
export { default as HotelCircle } from './HotelCircle.jsx'
export { default as Hotels } from './Hotels.jsx'
export { default as House } from './House.jsx'
export { default as Hybrid } from './Hybrid.jsx'
export { default as Icon } from './Icon.jsx'
export { default as Image } from './Image.jsx'
export { default as ImageWide } from './ImageWide.jsx'
export { default as Inclusive } from './Inclusive.jsx'
export { default as Information } from './Information.jsx'
export { default as InformationOutline } from './InformationOutline.jsx'
export { default as Instagram } from './Instagram.jsx'
export { default as Key } from './Key.jsx'
export { default as KingBed } from './KingBed.jsx'
export { default as Kitchen } from './Kitchen.jsx'
export { default as Kitchenette } from './Kitchenette.jsx'
export { default as Laptop } from './Laptop.jsx'
export { default as LateNight } from './LateNight.jsx'
export { default as License } from './License.jsx'
export { default as List } from './List.jsx'
export { default as ListContained } from './ListContained.jsx'
export { default as LocalBar } from './LocalBar.jsx'
export { default as Lock } from './Lock.jsx'
export { default as Logout } from './Logout.jsx'
export { default as Loyalty } from './Loyalty.jsx'
export { default as Luggage } from './Luggage.jsx'
export { default as LuggageNo } from './LuggageNo.jsx'
export { default as Manual } from './Manual.jsx'
export { default as Map } from './Map.jsx'
export { default as Mask } from './Mask.jsx'
export { default as Menu } from './Menu.jsx'
export { default as Microwave } from './Microwave.jsx'
export { default as Mileage } from './Mileage.jsx'
export { default as Minus } from './Minus.jsx'
export { default as MoreHorizontal } from './MoreHorizontal.jsx'
export { default as MoreVertical } from './MoreVertical.jsx'
export { default as MultiAirline } from './MultiAirline.jsx'
export { default as MultiOccupancy } from './MultiOccupancy.jsx'
export { default as Museum } from './Museum.jsx'
export { default as Navigation } from './Navigation.jsx'
export { default as NewRelease } from './NewRelease.jsx'
export { default as NoSeat } from './NoSeat.jsx'
export { default as Notification } from './Notification.jsx'
export { default as NotificationImportant } from './NotificationImportant.jsx'
export { default as Notifications } from './Notifications.jsx'
export { default as NotificationsAdd } from './NotificationsAdd.jsx'
export { default as OfflineBolt } from './OfflineBolt.jsx'
export { default as OnDemand } from './OnDemand.jsx'
export { default as Overnight } from './Overnight.jsx'
export { default as Palette } from './Palette.jsx'
export { default as ParentChild } from './ParentChild.jsx'
export { default as ParentChildCircle } from './ParentChildCircle.jsx'
export { default as Parking } from './Parking.jsx'
export { default as PersonalItem } from './PersonalItem.jsx'
export { default as Pets } from './Pets.jsx'
export { default as Phone } from './Phone.jsx'
export { default as PhoneActive } from './PhoneActive.jsx'
export { default as PhoneAgent } from './PhoneAgent.jsx'
export { default as PhoneLog } from './PhoneLog.jsx'
export { default as Picture } from './Picture.jsx'
export { default as PieChart } from './PieChart.jsx'
export { default as Pin } from './Pin.jsx'
export { default as PinAdd } from './PinAdd.jsx'
export { default as PinEdit } from './PinEdit.jsx'
export { default as Plus } from './Plus.jsx'
export { default as PlusOne } from './PlusOne.jsx'
export { default as Pool } from './Pool.jsx'
export { default as Power } from './Power.jsx'
export { default as Priceline } from './Priceline.jsx'
export { default as Printer } from './Printer.jsx'
export { default as Quilt } from './Quilt.jsx'
export { default as RadioChecked } from './RadioChecked.jsx'
export { default as RadioEmpty } from './RadioEmpty.jsx'
export { default as RadioMinus } from './RadioMinus.jsx'
export { default as RadioPlus } from './RadioPlus.jsx'
export { default as Railway } from './Railway.jsx'
export { default as Receipt } from './Receipt.jsx'
export { default as Refresh } from './Refresh.jsx'
export { default as Restaurant } from './Restaurant.jsx'
export { default as Ribbon } from './Ribbon.jsx'
export { default as RibbonAlt } from './RibbonAlt.jsx'
export { default as Road } from './Road.jsx'
export { default as RoomSize } from './RoomSize.jsx'
export { default as Rowing } from './Rowing.jsx'
export { default as SaveAlt } from './SaveAlt.jsx'
export { default as Search } from './Search.jsx'
export { default as SearchRecent } from './SearchRecent.jsx'
export { default as Seat } from './Seat.jsx'
export { default as SeatBusiness } from './SeatBusiness.jsx'
export { default as SeatEconomy } from './SeatEconomy.jsx'
export { default as Security } from './Security.jsx'
export { default as Share } from './Share.jsx'
export { default as Shuttle } from './Shuttle.jsx'
export { default as Sick } from './Sick.jsx'
export { default as SingleBed } from './SingleBed.jsx'
export { default as SingleOccupancy } from './SingleOccupancy.jsx'
export { default as Sites } from './Sites.jsx'
export { default as Sliders } from './Sliders.jsx'
export { default as Smartphone } from './Smartphone.jsx'
export { default as Smoking } from './Smoking.jsx'
export { default as SmokingAllowed } from './SmokingAllowed.jsx'
export { default as SoapBar } from './SoapBar.jsx'
export { default as Sort } from './Sort.jsx'
export { default as Spa } from './Spa.jsx'
export { default as Sparkle } from './Sparkle.jsx'
export { default as SplitTicket } from './SplitTicket.jsx'
export { default as SprayBottle } from './SprayBottle.jsx'
export { default as Star } from './Star.jsx'
export { default as StarHalf } from './StarHalf.jsx'
export { default as SteeringWheel } from './SteeringWheel.jsx'
export { default as Store } from './Store.jsx'
export { default as Subway } from './Subway.jsx'
export { default as SubwayAlt } from './SubwayAlt.jsx'
export { default as Success } from './Success.jsx'
export { default as SuccessOutline } from './SuccessOutline.jsx'
export { default as Sunrise } from './Sunrise.jsx'
export { default as Sustainability } from './Sustainability.jsx'
export { default as Swap } from './Swap.jsx'
export { default as SwapHorizontal } from './SwapHorizontal.jsx'
export { default as Sync } from './Sync.jsx'
export { default as TableChart } from './TableChart.jsx'
export { default as Task } from './Task.jsx'
export { default as TaskChecked } from './TaskChecked.jsx'
export { default as Taxi } from './Taxi.jsx'
export { default as Tea } from './Tea.jsx'
export { default as ThumbsDown } from './ThumbsDown.jsx'
export { default as ThumbsUp } from './ThumbsUp.jsx'
export { default as Timer } from './Timer.jsx'
export { default as Traffic } from './Traffic.jsx'
export { default as Train } from './Train.jsx'
export { default as Trash } from './Trash.jsx'
export { default as TravelExplore } from './TravelExplore.jsx'
export { default as TrendingUp } from './TrendingUp.jsx'
export { default as Trophy } from './Trophy.jsx'
export { default as Tune } from './Tune.jsx'
export { default as Twitter } from './Twitter.jsx'
export { default as Unlock } from './Unlock.jsx'
export { default as User } from './User.jsx'
export { default as UserAdd } from './UserAdd.jsx'
export { default as UserChecked } from './UserChecked.jsx'
export { default as UserCircle } from './UserCircle.jsx'
export { default as Verified } from './Verified.jsx'
export { default as ViewList } from './ViewList.jsx'
export { default as ViewModule } from './ViewModule.jsx'
export { default as Visibility } from './Visibility.jsx'
export { default as VisibilityOff } from './VisibilityOff.jsx'
export { default as Walk } from './Walk.jsx'
export { default as Warning } from './Warning.jsx'
export { default as WarningOutline } from './WarningOutline.jsx'
export { default as WashingHands } from './WashingHands.jsx'
export { default as WeatherHurricane } from './WeatherHurricane.jsx'
export { default as WeatherLightning } from './WeatherLightning.jsx'
export { default as WeatherRain } from './WeatherRain.jsx'
export { default as WeatherSnow } from './WeatherSnow.jsx'
export { default as WeatherSnowRain } from './WeatherSnowRain.jsx'
export { default as WeatherWindy } from './WeatherWindy.jsx'
export { default as Web } from './Web.jsx'
export { default as Whirlpool } from './Whirlpool.jsx'
export { default as Widgets } from './Widgets.jsx'
export { default as Wifi } from './Wifi.jsx'
export { default as Youtube } from './Youtube.jsx'
export { default as ZoomOut } from './ZoomOut.jsx'