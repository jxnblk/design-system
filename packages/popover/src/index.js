import React, { Component } from "react"
import ReactDOM from "react-dom"
import { Manager, Reference, Popper } from "react-popper"
import {Box, Flex, CloseButton, theme} from 'pcln-design-system'
import styled from "styled-components";
import DEFAULTS_MODIFIERS from "./helpers/defaultModifiers"

class PopOver extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isPopOverOpen: false
    }

    this.contentRef = React.createRef()
    this.triggerRef = React.createRef()
    this.handleToggle = this.handleToggle(this)
    this.handleOpen = this.handleOpen.bind(this)
    this.handleClose = this.handleClose.bind(this)
    this.setFocusToRef = this.setFocusToRef.bind(this)

    console.log(this.props)
  }

  handleToggle(isOpen) {
    if (isOpen) {
      this.handleClose()
    } else {
      this.handleOpen()
    }
  }

  handleClose() {
    this.setState( prevState => {
      if (prevState.isPopOverOpen) {
        return {
          isPopOverOpen: false
        }
      }
    }, () => {
      this.setFocusToRef(this.triggerRef)
    })
  }

  handleOpen() {
    this.setState( prevState => {
      if (!prevState.isPopOverOpen) {
        return {
          isPopOverOpen: true
        }
      }
    }, () => {
      this.setFocusToRef(this.contentRef)
    })
  }

  setFocusToRef(ref) {
    if (ref) {
      try {
        ref.current.focus()
      } catch {
        return false
      }
    }
  }

  render() {
    return (
      <Manager>
        <Reference>
          {({ ref }) => (
            // Need to be a span, because of ref forwarding limitations with functional components
            <span ref={ref}>
              {
                // Clone element to pass down toggle event so it can be used directly from children as needed
                React.cloneElement(this.props.children, {
                  'aria-label': 'Click for more information',
                  onClick: this.handleOpen,
                  ref: this.triggerRef //Currently ref only works with native element, if we use a DS core component it does not work.
                })
              }
            </span>
          )}
        </Reference>
        {
          this.state.isPopOverOpen &&
            ReactDOM.createPortal(
              <Popper
                positionFixed={true}
                modifiers={{ ...DEFAULTS_MODIFIERS }}
                {...this.props}
              >
                {({ placement, ref, style, arrowProps }) => (
                  // Need to be a native element, because of ref forwarding limitations with functional components
                  <span
                    /*
                     * NOTE: InnerRef has been depracted in V4 of styled components. Make sure to change this prop once we upgrade to styled components v4
                     * https://www.styled-components.com/docs/api#deprecated-innerref-prop
                     */
                    ref={ref}
                    style={style}
                    data-placement={placement}
                    aria-label={this.props.ariaLabel || 'Dialog Title'}
                    role='dialog'
                    describedby={`dialog-description-${this.props.id}`}
                  > 
                    <ContentContainer innerRef={this.contentRef} tabIndex='-1'>
                      {
                        this.props.allowClose &&
                        <Flex p={2} color='blue'>
                          <Box mx='auto' />
                          <CloseButton onClick={this.handleClose}/>
                        </Flex> 
                      }           
                      <Box id={`popover-description-${this.props.id}`}>
                        {
                          this.props.renderContent({
                            handleClose: this.handleClose
                          })
                        }
                      </Box>
                    </ContentContainer>
                  </span>
                )}
              </Popper>,
              // Append each instance of the Popover as portal directly to the body
              document.querySelector("body")
            )
        }
      </Manager>
    );
  }
}

const ContentContainer = styled.section`
  padding: ${({ theme }) => theme.space[2]}px;
  box-shadow: ${({ theme }) => theme.boxShadows[1]};
  font-size: ${({ theme }) => theme.fontSizes[0]}px;
  border-radius: ${({ theme }) => theme.radii[1]}px;
  box-sizing: border-box;
  background: ${({ theme, bg }) => theme.colors[bg]};
`

export default PopOver
