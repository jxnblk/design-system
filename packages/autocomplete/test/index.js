import React from 'react'
import us from 'us'
import { Autocomplete } from '../src'

describe('Autocomplete', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(
      <Autocomplete>
        <Autocomplete.Label>Test</Autocomplete.Label>
        <Autocomplete.Input />
        <Autocomplete.Menu>
          {us.STATES.map((state) => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders open', () => {
    const json = rendererCreateWithTheme(
      <Autocomplete defaultIsOpen={true}>
        <Autocomplete.Menu>
          {us.STATES.map((state) => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    const [menu] = json.children
    expect(menu.children.length > 0).toBe(true)
  })

  test('renders with highlightedIndex', () => {
    const json = rendererCreateWithTheme(
      <Autocomplete defaultIsOpen={true} defaultHighlightedIndex={0}>
        <Autocomplete.Menu>
          {us.STATES.map((state) => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    const [menu] = json.children
    const [firstItem] = menu.children
    expect(firstItem.props['data-highlighted']).toBe(true)
  })
})
