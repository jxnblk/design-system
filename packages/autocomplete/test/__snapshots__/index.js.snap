// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete renders 1`] = `
.c2 {
  margin-top: 4px;
  background-color: #fff;
  box-shadow: 0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16);
  border: 0;
  border-radius: 2px;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #c0cad5;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  font-size: 16px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  margin: 0;
  color: #4f6f8f;
  font-size: 10px;
  font-weight: 700;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 14px;
  }
}

<div
  aria-expanded={false}
  aria-haspopup="listbox"
  aria-labelledby="downshift-0-label"
  aria-owns={null}
  role="combobox"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <label
    className="c0"
    color="border.light"
    fontSize="10px"
    fontWeight="bold"
    htmlFor="downshift-0-input"
    id="downshift-0-label"
  >
    Test
  </label>
  <input
    aria-activedescendant={null}
    aria-autocomplete="list"
    aria-controls={null}
    aria-labelledby="downshift-0-label"
    autoComplete="off"
    className="c1"
    fontSize={
      Array [
        2,
        null,
        1,
      ]
    }
    id="downshift-0-input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
  <div
    aria-labelledby="downshift-0-label"
    className="c2 kLcFgh"
    height="256px"
    id="downshift-0-menu"
    role="listbox"
  />
</div>
`;
